public_website/views.py

```diff
@@ -749,24 +751,41 @@ def get(self, request):
     def post(self, request):
         form = ContactPageForm(request.POST, auto_id=False)
         if form.is_valid():
-            c = ContactUsSubmission()
-            c.name = form.cleaned_data.get('fullname')
-            c.emailid = form.cleaned_data.get('emailid')
-            c.subject = form.cleaned_data.get('subject')
-            c.message = form.cleaned_data.get('message')
-            c.save()
-
-            send_mail(
-                subject='[Contact] ' + form.cleaned_data.get('subject'),
-                message='',
-                html_message=form.cleaned_data.get('message') + '<br><br> Senders email: ' + form.cleaned_data.get('emailid'),
-                from_email='"Sawaliram" <mail@sawaliram.org>',
-                recipient_list=['mail.sawaliram@gmail.com'],
-                fail_silently=False,
-            )
 
-            messages.success(request, 'Your message has been sent! We will get back to you shortly')
-            return redirect('public_website:contact')
+            captcha_token = request.POST.get("g-recaptcha-response")
+            cap_url = "https://www.google.com/recaptcha/api/siteverify"
+            cap_secret = settings.GOOGLE_RECAPTCHA_SECRET_KEY
+
+            cap_data = {"secret":cap_secret,"response":captcha_token}
+
+            cap_server_response = requests.post(url=cap_url,data=cap_data)
+            cap_json = cap_server_response.json()
+            print(cap_json)
+
+            if cap_json['success']:
+
+                c = ContactUsSubmission()
+                c.name = form.cleaned_data.get('fullname')
+                c.emailid = form.cleaned_data.get('emailid')
+                c.subject = form.cleaned_data.get('subject')
+                c.message = form.cleaned_data.get('message')
+                c.save()
+
+                send_mail(
+                    subject='[Contact] ' + form.cleaned_data.get('subject'),
+                    message='',
+                    html_message=form.cleaned_data.get('message') + '<br><br> Senders email: ' + form.cleaned_data.get('emailid'),
+                    from_email='"Sawaliram" <mail@sawaliram.org>',
+                    recipient_list=['mail.sawaliram@gmail.com'],
+                    fail_silently=False,
+                )
+
+                messages.success(request, 'Your message has been sent! We will get back to you shortly')
```

Comment:
This line needs to be translated as well! It should be something like `messages.success(request, _('Your message has been sent! We will get back to you shortly')`. Similarly for a few other messages below, including the context setting.

(In general, if any string is in English, please mark it as translatable using the `_()` command: that way translators can identify and convert those strings into the regional language.

Reply:
I have resolved this issue

Reply:
Okay, looking good!